adaptive_breakpoints/test/adaptive_breakpoints_test.dart

```diff
@@ -39,249 +39,436 @@ void main() {
-        color: AdaptiveWindowType.m > AdaptiveWindowType.xl
-            ? trueColor
-            : falseColor,
-      )),
-    );
-
-    box = tester.widget(find.byType(Container));
-    expect((box as Container).color, falseColor);
-
-    await tester.pumpWidget(
-      MaterialApp(
-          home: Container(
-        color: AdaptiveWindowType.xl > AdaptiveWindowType.xl
-            ? trueColor
-            : falseColor,
-      )),
-    );
-
-    box = tester.widget(find.byType(Container));
-    expect((box as Container).color, falseColor);
+          (WidgetTester tester) async {
+        tester.binding.window.physicalSizeTestValue = const Size(399.5, 600);
+        tester.binding.window.devicePixelRatioTestValue = 1.0;
+
+        await tester.pumpWidget(
+          MaterialApp(
+              home: Container(
+                color: AdaptiveWindowType.xs > AdaptiveWindowType.xl
+                    ? trueColor
+                    : falseColor,
+              )),
+        );
+        var box = tester.widget(find.byType(Container));
+        expect((box as Container).color, falseColor);
+
+        await tester.pumpWidget(
+          MaterialApp(
+              home: Container(
+                color: AdaptiveWindowType.md > AdaptiveWindowType.xl
+                    ? trueColor
+                    : falseColor,
+              )),
+        );
+
+        box = tester.widget(find.byType(Container));
+        expect((box as Container).color, falseColor);
+
+        await tester.pumpWidget(
+          MaterialApp(
+              home: Container(
+                color: AdaptiveWindowType.xl > AdaptiveWindowType.xl
+                    ? trueColor
+                    : falseColor,
+              )),
+        );
+
+        box = tester.widget(find.byType(Container));
+        expect((box as Container).color, falseColor);
+
+        await tester.pumpWidget(
+          MaterialApp(
+              home: Container(
+                color: AdaptiveWindowType.lg > AdaptiveWindowType.sm
+                    ? trueColor
+                    : falseColor,
+              )),
+        );
+
+        box = tester.widget(find.byType(Container));
+        expect((box as Container).color, trueColor);
+      });
+}
 
-    await tester.pumpWidget(
-      MaterialApp(
-          home: Container(
-        color: AdaptiveWindowType.l > AdaptiveWindowType.s
-            ? trueColor
-            : falseColor,
-      )),
+/// [AdaptiveContainer] lets you create a [Container] with adaptive constraints.
+///
+/// The AdaptiveContainer does everything a normal container does but with
+/// adaptive constraints. For more information go to one of the links below.
+///
+/// https://api.flutter.dev/flutter/widgets/Container-class.html
+/// https://material.io/design/layout/responsive-layout-grid.html#breakpoints
+///
+/// This class is useful whenever you want a container to only be active in
+/// certain [AdaptiveWindowType].
+class _AdaptiveContainer extends StatelessWidget {
```

Comment:
Instead of recreating the _AdaptiveContainer in the tests, could you modify/simplify the tests to purely test the adaptive breakpoints functionality instead?

That way the tests that rely on `AdaptiveContainer` can instead move to the new package you create?

Reply:
Good idea